cpp/src/lib/osd_info.cc

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright (C) 2016 iNuron NV
+
+This file is part of Open vStorage Open Source Edition (OSE), as available from
+
+
+    http://www.openvstorage.org and
+    http://www.openvstorage.com.
+
+This file is free software; you can redistribute it and/or modify it
+under the terms of the GNU Affero General Public License v3 (GNU AGPLv3)
+as published by the Free Software Foundation, in version 3 as it comes
+in the <LICENSE.txt> file of the Open vStorage OSE distribution.
+
+Open vStorage is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY of any kind.
+*/
+#include "stuff.h"
+#include "osd_info.h"
+#include "llio.h"
+#include <iostream>
+#include "alba_logger.h"
+namespace alba {
+namespace llio {
+template <> void from(message &m, proxy_protocol::OsdInfo &info) {
+  uint8_t version;
+  from(m, version);
+  assert(version == 3);
+  std::string inner_s;
+  from(m, inner_s);
+  std::vector<char> inner_v(inner_s.begin(), inner_s.end());
+  message inner(inner_v);
+
+  uint8_t kind;
+  from(inner, kind);
+  assert(kind == 1);
+  from(inner, info.ips);
+  from(inner, info.port);
+  from(inner, info.use_tls);
+  from(inner, info.use_rdma);
+}
+
+template <> void from(message &m, proxy_protocol::OsdCapabilities &caps) {
+  uint32_t size;
+  from(m, size);
+  for (uint i = 0; i < size; i++) {
+    uint8_t version;
+    from(m, version);
+    assert(version == 1);
+    uint8_t tag;
+    from(m, tag);
+    switch (tag) { case 1: { }; break; // for now, we don't care
```

Comment:
better put an `assert (tag = 3)` here then as otherwise you could get unexpected results?